### DISCUSSION TITLE
Using context in lexer?

### DISCUSSION THREAD

**Discussion by quenbyako** [alecthomas/participle#223] [Q&A]
So, i using my custom lexer, it far reminds [Rob Pike's lexer](https://www.youtube.com/watch?v=HxaD_trXwRE) (really far, but mechanics are still pretty same).

I worry, that if parser fails parsing any input, my lexer's goroutine is still can't be stopped, cause it doesn't understand, does the parsing still going, or is it cancelled...

So question is, am i able somehow to get context of parsing inside lexers? If not, is it a good idea to implement this feature and create pull request?

**Comment by alecthomas** [comment]
If the parser returns an error, you should cancel any lexing goroutines.

**Reply by quenbyako** [reply]
yes, but how lexer could understand, that parsing was failed?
